collector/container.go

```diff
@@ -254,53 +255,6 @@ func (c *ContainerMetricsCollector) collect(ch chan<- prometheus.Metric) (*prome
 			containerIdWithPrefix,
 		)
 
-		if len(cstats.Network) == 0 {
```

Comment:
Do we need backward compatibility with docker runtime using this manner of querying for network stats? Not sure if HNS query is going to work on docker runtime and I didn't get a chance to test it as of yet.

Reply:
Yes we need backwards compat for awhile.  Docker is still using v1 api's so for folks that are using windows exporter without Kubernetes (aka just docker installed) will need to be supported.

Since the network in Docker uses v1, I am not sure if the stats will work from v2 api.  In kubernetes, the CNI's had all transitioned to using v2 api's so this wasn't a concern. 

We will need to test if the calls work, if not come up with a way make the distinction.

Reply:
Got it, will make the change test on both container runtimes and make sure the logic supports both.

Reply:
Despite this code removal, should be good here for both Docker and containerd. 

If necessary I am willling to make the change to support the v1 api calls removed here, but I don't think its necessary as I have tested on both Docker and containerd and metrics pull through successfully. See PR description for more information regarding the environment I used during testing.

Reply:
ah interesting!  I had a suspicion that this would work for both but wasn't sure.  I think the v2 api's have been released for long enough that we can drop the support 👍

Reply:
just had a thought, did you test this with docker outside of k8s?  I think in K8s the cni's are all using v2 apis

Reply:
@jsturtevant I have just concluded tests on a bare metal Windows host (not via Kubernetes) and metrics are showing as expected with no further changes. We should be good. 

See snippet of output of Prometheus metrics endpoint
```
# HELP windows_container_network_transmit_bytes_total Bytes Sent on Interface
# TYPE windows_container_network_transmit_bytes_total counter
windows_container_network_transmit_bytes_total{container_id="docker://062ac7f1925dd3a7054089b05bf9b2b8bdd4920d89fe3b944584e1dbf7e98065",interface="87AD7194-4445-4D5B-869F-5B6AF713C121"} 15745
```

Environment used in testing
- Docker Engine: 24.0.2
- OS Version: Windows Server 2022 Datacenter